### PR TITLE
Feat/issue 7661

### PR DESCRIPTION
**Pull request by Khushboo-dev-cpp**
### What does the PR do

Partially addresses #9302 

### StatusQ checklist

- [ ] add documentation if necessary (new component, new feature)
- [ ] update sandbox app
  - in case of new component, add new component page
  - in case of new features, add variation to existing component page
  - nice to have: add it to the demo application as well
- [ ] test changes in both light and dark theme?

### Screenshot of functionality (including design for comparison)

- [ ] I've checked the design and this PR matches it

1. AssetsView & Main Wallet Navigation Page

https://user-images.githubusercontent.com/60327365/211562516-cb7bb1ea-8231-45db-987f-225db70f636d.mov

### CODE REVIEW

**Review by alaibe — APPROVED** [review]
super cool!

**Review by stefandunca — APPROVED** [review]
Significant improvement for the wallet loading! Please check my inline note.

File: ui/imports/shared/views/LoadingGraphView.qml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    sourceComponent: Item {
+        LoadingComponent {
+            id: loadingComp
+            anchors.fill: parent
+            visible: false
+        }
+
+        Image {
+            id: mask
+            source: "file:/home/khushboo/Documents/Vector2.png"
```
[the remark this answers is not in the document]
Comment (Khushboo-dev-cpp) [reply ui/imports/shared/views/LoadingGraphView.qml]: thanks for catching that, Iv moved it to our shared assets folder!

File: src/app/modules/main/wallet_section/view.nim [hunk 2]
```diff
@@ -74,6 +85,10 @@ QtObject:
     self.currentCurrency = currency
     self.currentCurrencyChanged()
 
+  proc setTokensLoading*(self: View, loading: bool) =
+    self.tokensLoading = loading
```
Comment (MishkaRogachev) [inline src/app/modules/main/wallet_section/view.nim:89]:
```
if self.tokensLoading == loading:
    return
```

**Review by MishkaRogachev — APPROVED** [review]
Supercool animation! Just a few questions you are free to ignore

File: ui/imports/shared/controls/StyledTextEditWithLoadingState.qml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+    color: loading ? "transparent" : customColor
+
+    Loader {
+        anchors.left: parent.left
+        anchors.verticalCenter: parent.verticalCenter
+        active: root.loading
+        sourceComponent: LoadingComponent {
+            anchors.centerIn: parent
+            radius: textMetrics.font.pixelSize === 15 ? 4 : 8
```
[the remark this answers is not in the document]
Comment (Khushboo-dev-cpp) [reply ui/imports/shared/controls/StyledTextEditWithLoadingState.qml:21]: made it <= instead :)

File: ui/imports/shared/controls/StyledTextEditWithLoadingState.qml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+StyledTextEdit {
+    id: root
+
+    property bool loading: false
+    property color customColor: Style.current.textColor
+
+    color: loading ? "transparent" : customColor
+
+    Loader {
+        anchors.left: parent.left
```
[the remark this answers is not in the document]
Comment (Khushboo-dev-cpp) [reply ui/imports/shared/controls/StyledTextEditWithLoadingState.qml:16]: it was not needed, removed the anchors.centerIn

File: ui/imports/shared/views/AssetsView.qml [hunk 5]
```diff
@@ -32,16 +32,31 @@ Item {
...
-        model: SortFilterProxyModel {
-            sourceModel: account.assets
-            filters: [
-                ExpressionFilter {
-                    expression: visibleForNetworkWithPositiveBalance
-                }
-            ]
+        model: RootStore.tokensLoading ? 25 : filteredModel
+        delegate: RootStore.tokensLoading ? loadingTokenDelegate : tokenDelegate
+        ScrollBar.vertical: ScrollBar { policy: ScrollBar.AsNeeded }
```
[the remark this answers is not in the document]
Comment (Khushboo-dev-cpp) [reply ui/imports/shared/views/AssetsView.qml:37]: you are right, using the correct component now

**Pull request merged**